File: projects/04/01/rectangle_area.rb
#! /usr/bin/ruby
# rectangle_area.rb computes the area of a rectangle
# 
# Input: the length and width of the rectangle
# Precondition: the length and width are positive
# Output: the area of the rectangle.
#
# Begun by: Dr. Nelesen, for CS 214 at Calvin College.
# Completed by: Zach Wibbenmeyer
# Date: February 12, 2016
###############################################################

# function rectangleArea returns a rectangle's area, given its length and width
# @param: l, w, numbers
# Precondition: l and w are positive
# Return: the area of the rectangle

def rectangleArea(l, w)
	#Calculate the area
	l * w
end

# Main program

if __FILE__ == $0

	#Prompt the user for the length
	print "To calculate the area of a rectangle, enter the length: "
	length = gets.chomp.to_f
	
	#Prompt the user for the width
	print "Now, enter the width: "
	width = gets.chomp.to_f

	#Print the area
	print "Area is: "
	#Call the rectangleArea function
	puts rectangleArea(length, width)
end

File: projects/04/01/rectangle_area.java
/* rectangle_area.java computes the area of a rectangle.
 *
 * Input: The two lengths of the rectangle.
 * Precondition: The lengths are positive numbers.
 * Output: The area of the rectangle.
 *
 * Begun by: Dr. Nelesen, for CS 214, at Calvin College.
 * Completed by: Zach Wibbenmeyer
 * Date: February 11, 2016
 **********************************************************/

import java.util.Scanner;   // include class for scanner

public class rectangle_area {

	/* function rectangleArea() computes a rectangle's area
	 * @param: l, w, doubles
	 * Precondition: l and w are not negative
	 * Returns: the area of the rectangle
	 */
	public static double rectangleArea(double l, double w) {
		return l * w;		// return the area
	}

   // main program
   public static void main(String[] args) {
      
      // Create a connection named keyboard to standard in
      Scanner keyboard = new Scanner(System.in);

      // prompt for the width
      System.out.println("To compute the area of the rectangle, enter the width: ");

      // Get the first length from the user
      double width = keyboard.nextDouble();

      // Prompt for the length
      System.out.println("Now enter the length: ");

      // Get the length from the user
      double length = keyboard.nextDouble();

      // output the area
      System.out.println("The area is " + rectangleArea(width, length));
   }
}
      
